File: src/main/java/com/anatartari/tatamecontrolapi/infra/database/entity/AddressEntity.java
package com.anatartari.tatamecontrolapi.infra.database.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.sql.Struct;

@Getter
@Setter
@Entity
@Table(name = "address")
public class AddressEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String street;

    @Column(nullable = false)
    private String neighborhood;

    @Column(nullable = false)
    private String cep;

    private String city;
    private String state;
    private String number;
    private String complement;
}

File: src/address/entities/address.entity.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity()
export class Address {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ length: 255 })
    street: string;

    @Column({ length: 255 })
    neighborhood: string;

    @Column({ length: 20 })
    cep: string;

    @Column({ length: 100, nullable: true })
    city?: string;

    @Column({ length: 50, nullable: true })
    state?: string;

    @Column({ length: 20, nullable: true })
    number?: string;

    @Column({ length: 100, nullable: true })
    complement?: string;
}